storage/src/main/java/io/delta/storage/LogStore.java

```diff
@@ -101,7 +101,7 @@ public abstract void write(
         Path path,
         Iterator<String> actions,
         Boolean overwrite,
-        Configuration hadoopConf) throws IOException;
+        Configuration hadoopConf) throws Exception;
```

Comment:
what are you throwing that requires you to change the public interface? let's avoid changing it if we can

Reply:
I couldn't avoid it. Exception is thrown by the ThreadUtils.runInNewThread()
  if (!exceptionHolder.isEmpty()) {
      Exception realException = exceptionHolder.get(0);

Reply:
commented below. Throw a `RuntimeException` instead.